tests/app/clients/test_zendesk_sell.py

```diff
@@ -296,44 +320,58 @@ def match_json(request):
             assert not deal_id
 
 
-@pytest.mark.parametrize('is_go_live', [False, True])
+@pytest.mark.parametrize('is_go_live,existing_contact_id', [
+    (False, None),
+    (False, '1'),
+    (True, None)
+])
 def test_create_service_or_go_live_contact_fail(
         notify_api: Flask,
         sample_service: Service,
         mocker: MockFixture,
-        is_go_live: bool
+        is_go_live: bool,
+        existing_contact_id: Optional[str]
 ):
 
     upsert_contact_mock = mocker.patch('app.user.rest.ZenDeskSell.upsert_contact', return_value=(None, False))
+    convert_lead_to_contact_mock = \
+        mocker.patch('app.user.rest.ZenDeskSell.convert_lead_to_contact', return_value=existing_contact_id)
+
     with notify_api.app_context():
         if is_go_live:
             assert not ZenDeskSell().send_go_live_service(sample_service, sample_service.users[0])
+            upsert_contact_mock.assert_called_once_with(sample_service.users[0], existing_contact_id)
         else:
             assert not ZenDeskSell().send_create_service(sample_service, sample_service.users[0])
-
-        upsert_contact_mock.assert_called_once_with(sample_service.users[0])
+            convert_lead_to_contact_mock.assert_called_once_with(sample_service.users[0])
+            upsert_contact_mock.assert_called_once_with(sample_service.users[0], existing_contact_id)
 
 
-@pytest.mark.parametrize('is_go_live', [False, True])
+@pytest.mark.parametrize('is_go_live,existing_contact_id', [
+    (False, None),
+    (False, '1'),
+    (True, None)
+])
 def test_create_service_or_go_live_deal_fail(
         notify_api: Flask,
         sample_service: Service,
         mocker: MockFixture,
-        is_go_live: bool
+        is_go_live: bool,
+        existing_contact_id: Optional[str]
 ):
 
     with requests_mock.mock() as rmock:
-        contact_id = 1
+        contact_id = existing_contact_id or '1'
```

Comment:
Curious about this because `contact_id` will always be `1` since you pass `None`

Reply:
my mistake it should be different